akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -2184,23 +2187,73 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
     }
   }
 
+  private def createGossipMessages(
+      keys: immutable.Iterable[KeyId],
+      sendBack: Boolean,
+      fromSystemUid: Option[Long]): Vector[Gossip] = {
+    // The sizes doesn't have to be exact, rather error on too small messages. The serializer is also
+    // compressing the Gossip message.
+    val maxMessageSize = payloadSizeAggregator.maxFrameSize - 128
+
+    var messages = Vector.empty[Gossip]
+    var collectedEntries = Vector.empty[(KeyId, DataEnvelope)]
```

Comment:
Not sure how performance sensitive this is (or how big datasets we expect), perhaps worth using vector builder for at least `messages`?

Reply:
I changed to builder for `collectedEntries`. There can be rather many entires before filling up a message.
Kept `messages` as plain Vector since it's not expected to be split up in that many messages.

Reply:
that change was in d7a4e67 if you want to take an extra look